src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -0,0 +1,355 @@
+  }
+
+  void PeakIntegrator::estimateBackground(const MSSpectrum& spectrum, const double& left, const double& right)
+  {
+    estimateBackground_(spectrum, left, right);
+  }
+
+  void PeakIntegrator::integratePeak(const MSChromatogram& chromatogram, const double& left, const double& right)
+  {
+    integratePeak_(chromatogram, left, right);
+  }
+
+  void PeakIntegrator::integratePeak(const MSSpectrum& spectrum, const double& left, const double& right)
+  {
+    integratePeak_(spectrum, left, right);
+  }
+
+  double PeakIntegrator::simpson(MSChromatogram::ConstIterator it_begin, MSChromatogram::ConstIterator it_end) const
+  {
+    return simpson_(it_begin, it_end);
+  }
+
+  double PeakIntegrator::simpson(MSSpectrum::ConstIterator it_begin, MSSpectrum::ConstIterator it_end) const
+  {
+    return simpson_(it_begin, it_end);
+  }
+
+  void PeakIntegrator::calculatePeakShapeMetrics(const MSChromatogram& chromatogram, const double& left, const double& right)
+  {
+    calculatePeakShapeMetrics_(chromatogram, left, right);
+  }
+
+  void PeakIntegrator::calculatePeakShapeMetrics(const MSSpectrum& spectrum, const double& left, const double& right)
+  {
+    calculatePeakShapeMetrics_(spectrum, left, right);
+  }
+
+  double PeakIntegrator::getPeakArea() const { return peak_area_; }
+  double PeakIntegrator::getPeakHeight() const { return peak_height_; }
+  double PeakIntegrator::getPeakApexPos() const { return peak_apex_pos_; }
+  double PeakIntegrator::getBackgroundHeight() const { return background_height_; }
+  double PeakIntegrator::getBackgroundArea() const { return background_area_; }
+  double PeakIntegrator::getWidthAt5() const { return width_at_5_; }
+  double PeakIntegrator::getWidthAt10() const { return width_at_10_; }
+  double PeakIntegrator::getWidthAt50() const { return width_at_50_; }
+  double PeakIntegrator::getStartTimeAt5() const { return start_time_at_5_; }
+  double PeakIntegrator::getStartTimeAt10() const { return start_time_at_10_; }
+  double PeakIntegrator::getStartTimeAt50() const { return start_time_at_50_; }
+  double PeakIntegrator::getEndTimeAt5() const { return end_time_at_5_; }
+  double PeakIntegrator::getEndTimeAt10() const { return end_time_at_10_; }
+  double PeakIntegrator::getEndTimeAt50() const { return end_time_at_50_; }
+  double PeakIntegrator::getTotalWidth() const { return total_width_; }
+  double PeakIntegrator::getTailingFactor() const { return tailing_factor_; }
+  double PeakIntegrator::getAsymmetryFactor() const { return asymmetry_factor_; }
+  double PeakIntegrator::getBaselineDeltaToHeight() const { return baseline_delta_2_height_; }
+  double PeakIntegrator::getSlopeOfBaseline() const { return slope_of_baseline_; }
+  Int PeakIntegrator::getPointsAcrossBaseline() const { return points_across_baseline_; }
+  Int PeakIntegrator::getPointsAcrossHalfHeight() const { return points_across_half_height_; }
+
+  const std::map<String, double> PeakIntegrator::getPeakShapeMetrics() const
```

Comment:
I am not a big fan of this approach, why not explicitely encode this in a struct and remove all these methods, basically have

```
struct PeakShapeMetrics
{
double peak_area;
...
}

PeakShapeMetrics PeakIntegrator::getPeakShapeMetrics() const 
```

to me it seems that would get rid of a lot of unnecessary code. I only really see advantages to this approach, honestly.